From 13cd62bda3550fb1f5dfc7993781acd0f9b747bc Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 28 Mar 2022 20:26:31 +0000
Subject: [PATCH 1064/1077] MANTIS 0059165 MANTIS 0059165 bugfix: Call CUSS 1.5
 Shortcuts without extension from Platform config

massai_systools.lib
V1.0.0.44
- fix calculation of timestamp + duration
- use getLocalTimestamp for process start/stop time
- add mSysTimeOut for formatted output
- add boost unit test for addition and subtraction


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1719 90b65887-3827-0410-9a23-83215b262276
---
 .../Interfaces/inc/systools/mSysTime.h         |  1 +
 .../Interfaces/inc/systools/mSysTimeOut.h      | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)
 create mode 100644 SelfServiceCommon/Interfaces/inc/systools/mSysTimeOut.h

diff --git a/SelfServiceCommon/Interfaces/inc/systools/mSysTime.h b/SelfServiceCommon/Interfaces/inc/systools/mSysTime.h
index 8407e3ab..1d980c8d 100644
--- a/SelfServiceCommon/Interfaces/inc/systools/mSysTime.h
+++ b/SelfServiceCommon/Interfaces/inc/systools/mSysTime.h
@@ -59,6 +59,7 @@ namespace mSysTime
   Duration  operator-(const Duration&  left, const Duration&  right);
   Duration  operator+(const Duration&  left, const Duration&  right);
   Timestamp operator+(const Timestamp& left, const Duration&  right);
+  Timestamp operator-(const Timestamp& left, const Duration&  right);
 }
 
 #endif
diff --git a/SelfServiceCommon/Interfaces/inc/systools/mSysTimeOut.h b/SelfServiceCommon/Interfaces/inc/systools/mSysTimeOut.h
new file mode 100644
index 00000000..9bdeafc6
--- /dev/null
+++ b/SelfServiceCommon/Interfaces/inc/systools/mSysTimeOut.h
@@ -0,0 +1,18 @@
+// time; current time, system up time, sleep
+
+#ifndef MSYSTIMEOUT_H_INCLUDED
+#define MSYSTIMEOUT_H_INCLUDED
+#ifdef SHOW_INCLUDE_FILES
+#pragma message("including: " __FILE__)
+#endif
+
+#include <iostream>
+#include "systools/mSysTime.h"
+
+namespace std
+{
+  std::ostream& operator<<(std::ostream& out, const mSysTime::Timestamp& time);
+  std::ostream& operator<<(std::ostream& out, const mSysTime::Duration& duration);
+}
+
+#endif
-- 
2.41.0.windows.1

